extensions/standard-processors/processors/SplitText.cpp

```diff
@@ -0,0 +1,383 @@
+      }
+
+      finalizeFragmentOffset(current_fragment);
+      return current_fragment;
+    }
+
+    ++current_fragment.processed_line_count;
+    addLineToFragment(current_fragment, *line);
+    if (split_text_config_.line_split_count == current_fragment.processed_line_count) {
+      finalizeFragmentOffset(current_fragment);
+      return current_fragment;
+    }
+  }
+
+  if (current_fragment.fragment_size > 0) {
+    finalizeFragmentOffset(current_fragment);
+    return current_fragment;
+  }
+  return std::nullopt;
+}
+
+}  // namespace detail
+
+void SplitText::initialize() {
+  setSupportedProperties(Properties);
+  setSupportedRelationships(Relationships);
+}
+
+void SplitText::onSchedule(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSessionFactory>& /*sessionFactory*/) {
+  gsl_Expects(context);
+  split_text_config_.line_split_count = utils::getRequiredPropertyOrThrow<uint64_t>(*context, LineSplitCount.name);
+  logger_->log_debug("SplitText line split count: {}", split_text_config_.line_split_count);
+  auto max_fragment_data_size_value = context->getProperty<core::DataSizeValue>(MaximumFragmentSize);
+  if (max_fragment_data_size_value) {
+    split_text_config_.maximum_fragment_size = max_fragment_data_size_value->getValue();
+    logger_->log_debug("SplitText maximum fragment size: {}", split_text_config_.maximum_fragment_size.value());
+  }
+  if (split_text_config_.maximum_fragment_size && split_text_config_.maximum_fragment_size.value() == 0) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Maximum Fragment Size cannot be 0!");
+  }
+  if (split_text_config_.line_split_count == 0 && !split_text_config_.maximum_fragment_size) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Line Split Count is set to 0, but Maximum Fragment Size is not set!");
+  }
+  split_text_config_.header_line_count = utils::getRequiredPropertyOrThrow<uint64_t>(*context, HeaderLineCount.name);
+  logger_->log_debug("SplitText header line count: {}", split_text_config_.header_line_count);
+  split_text_config_.header_line_marker_characters = context->getProperty(HeaderLineMarkerCharacters);
+  if (split_text_config_.header_line_marker_characters && split_text_config_.header_line_marker_characters->size() >= detail::SPLIT_TEXT_BUFFER_SIZE) {
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, fmt::format("SplitText header line marker characters length is larger than the maximum allowed: {}", detail::SPLIT_TEXT_BUFFER_SIZE));
+  }
+  if (split_text_config_.header_line_marker_characters) {
+    logger_->log_debug("SplitText header line marker characters were set: {}", *split_text_config_.header_line_marker_characters);
+  }
+  split_text_config_.remove_trailing_new_lines = utils::getRequiredPropertyOrThrow<bool>(*context, RemoveTrailingNewlines.name);
+  logger_->log_debug("SplitText should remove trailing new lines: {}", split_text_config_.remove_trailing_new_lines ? "true" : "false");
```

Comment:
`fmt::format` can do this automagically:
```suggestion
  logger_->log_debug("SplitText should remove trailing new lines: {}", split_text_config_.remove_trailing_new_lines);
```

Reply:
Updated in fccf152353ae7c25a8d4944b8b0e3add15254984